Description: Initialize the client at first connection if not done yet.
Bug-Debian: http://bugs.debian.org/100375
Forwarded: no
Author: Philip Kendall <pak21@srcf.ucam.org>
Last-Update: 2010-07-01

--- a/lib/Rconnect.c
+++ b/lib/Rconnect.c
@@ -75,6 +75,10 @@
 		return connect(s, name, namelen);
 	}
 
+#if SOCKS_CLIENT /* may be called before normal init, log to right place */
+	clientinit();
+#endif     
+   
 	slog(LOG_DEBUG, "%s: %s",
 	function, sockaddr2string(name, namestring, sizeof(namestring)));
 
